### PR TITLE
Build progress UX

### PR DESCRIPTION
**Pull request by Naatan**
https://www.pivotaltracker.com/story/show/175563258

 https://www.pivotaltracker.com/story/show/175563258

### CODE REVIEW

**Comment by mdrohmann** [comment]
Sample output: ```
╔═══════════════════╗
║ Installing Bundle ║
╚═══════════════════╝
Updating Runtime
Changes to your runtime may require some dependencies to be rebuilt.

No bundle version specified, choosing version 1.00

Catalyst Bundle includes 55 dependencies
  ├─ Catalyst-Plugin-URI (127 dependencies)
  ├─ Catalyst-ActionRole-MatchRequestMethod (120 dependencies)
  ├─ Catalyst-Plugin-SmartURI (116 dependencies)
  ├─ Catalyst-Plugin-Session-State-Cookie (134 dependencies)
  ├─ CatalystX-AppBuilder (115 dependencies)
  ├─ Catalyst-Devel (142 dependencies)
  ├─ Catalyst-Component-InstancePerContext (115 dependencies)
  ├─ Catalyst-ActionRole-ACL (115 dependencies)
  ├─ CatalystX-DebugFilter (41 dependencies)
  ├─ Catalyst-Plugin-SubRequest (115 dependencies)
  ├─ Catalyst-Plugin-Log-Log4perlSimple (31 dependencies)
  ├─ Catalyst-ActionRole-DetachOnDie (124 dependencies)
  ├─ Catalyst-Plugin-FormValidator (130 dependencies)
  ├─ Catalyst-Action-FromPSGI (115 dependencies)
  ├─ CatalystX-LeakChecker (120 dependencies)
  ├─ Catalyst-Plugin-MemoryUsage (48 dependencies)
  ├─ Catalyst-Candy (116 dependencies)
  ├─ Catalyst-Plugin-StatusMessage (116 dependencies)
  ├─ Catalyst-Plugin-Authentication (134 dependencies)
  ├─ Catalyst-Plugin-Session-Store-Delegate (134 dependencies)
  ├─ Catalyst-ActionRole-RequireSSL (124 dependencies)
  ├─ Catalyst-View-JSON-PerRequest (134 dependencies)
  ├─ Catalyst-Controller-ActionRole (118 dependencies)
  ├─ Catalyst-Plugin-InjectionHelpers (115 dependencies)
  ├─ Catalyst-Runtime (114 dependencies)
  ├─ Catalyst-Model-Adaptor (115 dependencies)
  ├─ Catalyst-Plugin-I18N (116 dependencies)
  ├─ Catalyst-Plugin-Authorization-Roles (137 dependencies)
  ├─ CatalystX-Profile (129 dependencies)
  ├─ Catalyst-Action-REST (116 dependencies)
  ├─ CatalystX-Routes (117 dependencies)
  ├─ Catalyst-Controller-AutoAssets (135 dependencies)
  ├─ Catalyst-Plugin-Static-Simple (119 dependencies)
  ├─ Catalyst-Plugin-Cache (115 dependencies)
  ├─ Catalyst-View-JSON (115 dependencies)
  ├─ Catalyst-Plugin-Params-Nested (1 dependencies)
  ├─ CatalystX-REPL (131 dependencies)
  ├─ Catalyst-View-TT (119 dependencies)
  ├─ Catalyst-Plugin-StackTrace (115 dependencies)
  ├─ Catalyst-Plugin-Session (133 dependencies)
  ├─ Catalyst-Plugin-Session-Store-FastMmap (135 dependencies)
  ├─ Catalyst-Component-ACCEPT_CONTEXT (116 dependencies)
  ├─ CatalystX-Component-Traits (119 dependencies)
  ├─ Catalyst-View-Email (136 dependencies)
  ├─ Catalyst-Action-RenderView (119 dependencies)
  ├─ CatalystX-InjectComponent (123 dependencies)
  ├─ Catalyst-Manual (1 dependencies)
  ├─ Catalyst-ActionRole-QueryParameter (125 dependencies)
  ├─ CatalystX-RoleApplicator (116 dependencies)
  ├─ Catalyst-DispatchType-Regex (115 dependencies)
  ├─ Catalyst-Plugin-ConfigLoader (118 dependencies)
  ├─ Catalyst-Plugin-Authorization-ACL (116 dependencies)
  ├─ Catalyst-Plugin-SimpleMessage (116 dependencies)
  ├─ CatalystX-Controller-Auth (1 dependencies)
  └─ Catalyst-Plugin-CustomErrorMessage (22 dependencies)

Packages are automatically added to your runtime.
Building Remotely  318 / 318       
```

**Comment by mdrohmann** [comment]
This is certainly a good use-case to consider when we think about refactoring the runtime activation.  I believe that a more event-based design with a life-cycle model, might be beneficial.  The `runtime.RuntimeMessageHandler` already moves in that direction, but I believe that we would profit from making such a concept more central, and not only for output messages.  Potentially, we want to add functionality to every bit of the activation process:  Eg., 
- print something as soon as the recipe is known
- update some struct every time a new artifact tarball is installed
- ...

This is just a 2-cents comment after implementing this story which was quite painful...

**Review by MDrakos — COMMENTED** [review]
Overall this looks good to me. I will leave the more in depth review to Nathan.

One thing I did notice is this seems to be missing the `Building Packages` and `Downloading New Packages` lines from the mock. I think we need to update those lines and substitute `packages` for `dependencies`.

File: pkg/platform/runtime/download.go [hunk 1]
```diff
@@ -83,7 +85,7 @@ type Downloader interface {
 	Download(artifacts []*HeadChefArtifact, d DownloadDirectoryProvider, progress *progress.Progress) (files map[string]*HeadChefArtifact, fail *failures.Failure)
 
 	// FetchArtifacts will fetch artifact
-	FetchArtifacts() (*FetchArtifactsResult, *failures.Failure)
+	FetchArtifacts(recipe *inventory_models.Recipe, project *mono_models.Project) (*FetchArtifactsResult, *failures.Failure)
```
Comment (mdrohmann) [inline pkg/platform/runtime/download.go:88]: I like this change! :+1:

**Review by mdrohmann — CHANGES_REQUESTED** [review]
LGTM.  I understand your changes now.  Just found an error in a locale string that I added myself.

**Pull request merged**